diff --git a/SConstruct b/SConstruct
index 3ebdff9..72d9900 100644
--- a/SConstruct
+++ b/SConstruct
@@ -19,6 +19,13 @@ except Exception:
 
 env.PrependENVPath("PATH", os.getenv("PATH"))
 
+# For windows compilation with nix
+platform_arg = ARGUMENTS.get("platform", None)
+if platform_arg == "windows":
+    env.Append(CPPPATH=[os.getenv("MCFGTHREADS_INCLUDE_DIR")])
+    env.Append(LIBPATH=[os.getenv("MCFGTHREADS_LIB_DIR"), os.getenv("MCFGTHREADS_DLL_DIR")])
+    env.Append(LIBS=['mcfgthread'])
+
 # Custom options and profile flags.
 customs = ["custom.py"]
 try:
